tests/unit-tests/ProductDetail.php

```diff
@@ -42,4 +43,98 @@ public function test_view_item_event() {
 		$this->assertEquals( $expected_data, $this->get_event_data(), 'Event data does not match expected data structure for view_item (product_detail()) event' );
 	}
 
+	/**
+	 * Check that `WC_Google_Gtag_JS` registers
+	 * the `assets/js/ga-integration.js` script
+	 * as `woocommerce-google-analytics-integration` (`->script_handle`)
+	 * on `wp_enqueue_scripts` action.
+	 *
+	 * @return void
+	 */
+	public function test_variation_scripts_is_registered() {
+		$gtag = new WC_Google_Gtag_JS();
+
+		// Mimic WC action.
+		do_action( 'wp_enqueue_scripts' );
+
+		// Assert the handle is regeistered with the correct name, but not yet enqueued.
+		$this->assertEquals( 'woocommerce-google-analytics-integration', $gtag->script_handle, '`WC_Google_Gtag_JS->script_handle` is not equal `woocommerce-google-analytics-integration`' );
```

Comment:
There were some changes in `trunk` that have been merged back into this branch so the handle is no longer `$gtag->script_handle` and is now `$gtag->script_handle . '-ga-integration'`.

I just tested this test on a [new branch](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/273) after changing the script handle and the path below as it's now in a `build` subdirectory and they passed.

Really confusing though as if you look back at the previous commit on this PR those changes didn't exist on the branch when they first failed. 😕

Reply:
For context: 

The reason of this change is because no we are adding more than one script in `WC_Google_Gtag_JS`

`$this->script_handle . '-ga-integration'` (former  `$this->script_handle`) and `$this->script_handle . '-actions'`